fbw-a32nx/src/systems/fmgc/src/guidance/vnav/descent/TodGuidance.ts

```diff
@@ -44,61 +62,65 @@ export class TodGuidance {
 
   update(deltaTime: number) {
     this.updateTdReached(deltaTime);
-    this.updateTdPause(deltaTime);
+    if (NXDataStore.get('PAUSE_AT_TOD', 'DISABLED') === 'ENABLED') {
+      this.updateTdPause(deltaTime);
+    }
   }
 
   updateTdPause(deltaTime: number) {
-    if (this.cooldown <= 0 && NXDataStore.get('PAUSE_AT_TOD', 'DISABLED') === 'ENABLED') {
-      // Only watching if T/D pause untriggered + between flight phase CLB and CRZ
-      if (
+    // Only armed if all conditions met
+    this.tdArmed.setVar(
+      this.cooldown <= 0 &&
         !this.tdPaused &&
         this.observer.get().flightPhase >= FmgcFlightPhase.Climb &&
         this.observer.get().flightPhase <= FmgcFlightPhase.Cruise &&
-        Simplane.getAutoPilotAirspeedManaged()
+        Simplane.getAutoPilotAirspeedManaged(),
+    );
+
+    if (this.tdArmed) {
```

Comment:
Just a comment as I just tested this PR: I think it should be `if (this.tdArmed.getVar()) {` as I got spammed with Dialogs and the current condition is always true (as this.tdArmed is not undefined)

Reply:
I just pushed it as the change was minimal, feel free to revert if you don't like it 👍